src/parseTools.js

```diff
@@ -391,8 +397,13 @@ function makeSetValueImpl(ptr, pos, value, type) {
 
 function makeHEAPView(which, start, end) {
   const size = parseInt(which.replace('U', '').replace('F', '')) / 8;
-  const mod = size == 1 ? '' : ('>>' + Math.log2(size));
-  return `HEAP${which}.subarray((${start})${mod}, (${end})${mod})`;
+  if (MEMORY64) {
+    const mod = size == 1 ? '' : ('/' + size);
+    return `HEAP${which}.subarray((${start})${mod}, (${end})${mod})`;
+  } else {
+    const mod = size == 1 ? '' : ('>>' + Math.log2(size));
+    return `HEAP${which}.subarray((${start})${mod}, (${end})${mod})`;
+  }
```

Comment:
The last line of each branch here looks the same, right?    Can we simplify here to avoid duplication? e.g.:

```
if (MEMORY64) {
  mod = ..
} else {
  mod = ..
}
return `HEAP${which}.subarray((${start})${mod}, (${end})${mod})`;  
```

Reply:
Ok, will make this change.